pkg/cluster/deploystorage.go

```diff
@@ -118,15 +118,22 @@ func (m *manager) deployStorageTemplate(ctx context.Context) error {
 		m.clusterNSG(infraID, azureRegion),
 		m.clusterServicePrincipalRBAC(),
 		m.networkPrivateLinkService(azureRegion),
-		m.networkPublicIPAddress(azureRegion, infraID+"-pip-v4"),
 		m.networkInternalLoadBalancer(azureRegion),
-		m.networkPublicLoadBalancer(azureRegion),
 	}
 
-	if m.doc.OpenShiftCluster.Properties.IngressProfiles[0].Visibility == api.VisibilityPublic {
+	// We do not want to create a public IP or LB for clusters with user defined outbound types
+	if m.doc.OpenShiftCluster.Properties.NetworkProfile.OutboundType == api.OutboundTypeDefault {
```

Comment:
nit: this line is making me think we should have a third constant:

1. User Defined
2. (whatever it is we want to call what we do right now... public?)
3. Default (which may change over time)

This `if` makes a lot more sense if we use a constant like "public" instead of "default"

Reply:
+1 I don't think it makes sense to introduce a third constant at this time since we either want it user defined or default. I will rename the "default" to `OutboundTypeLoadbalancerRouting` since that will be more clear.